src/ocean/task/TaskPool.d

```diff
@@ -181,6 +185,9 @@ class TaskPool ( TaskT : Task ) : ObjectPool!(Task)
 
         foreach (task; tasks_iterator)
         {
+            enforce(!isSame(this.toItem(current_task), this.toItem(task)),
```

Comment:
Isn't it a programmer's error to end up in this situation? Is there any other way how would you end in this situation?